src/main/java/com/puppycrawl/tools/checkstyle/Checker.java

```diff
@@ -19,605 +19,13 @@
-        if (extensions == null) {
-            fileExtensions = null;
-        }
-        else {
-            fileExtensions = new String[extensions.length];
-            for (int i = 0; i < extensions.length; i++) {
-                final String extension = extensions[i];
-                if (CommonUtils.startsWithChar(extension, '.')) {
-                    fileExtensions[i] = extension;
-                }
-                else {
-                    fileExtensions[i] = "." + extension;
-                }
-            }
-        }
-    }
-
-    /**
-     * Sets the factory for creating submodules.
-     *
-     * @param moduleFactory the factory for creating FileSetChecks
-     */
-    public void setModuleFactory(ModuleFactory moduleFactory) {
-        this.moduleFactory = moduleFactory;
-    }
-
-    /**
-     * Sets locale country.
-     * @param localeCountry the country to report messages
-     */
-    public void setLocaleCountry(String localeCountry) {
-        this.localeCountry = localeCountry;
-    }
-
-    /**
-     * Sets locale language.
-     * @param localeLanguage the language to report messages
-     */
-    public void setLocaleLanguage(String localeLanguage) {
-        this.localeLanguage = localeLanguage;
-    }
-
-    /**
-     * Sets the severity level.  The string should be one of the names
-     * defined in the {@code SeverityLevel} class.
-     *
-     * @param severity  The new severity level
-     * @see SeverityLevel
-     */
-    public final void setSeverity(String severity) {
-        severityLevel = SeverityLevel.getInstance(severity);
-    }
-
-    /**
-     * Sets the classloader that is used to contextualize fileset checks.
-     * Some Check implementations will use that classloader to improve the
-     * quality of their reports, e.g. to load a class and then analyze it via
-     * reflection.
-     * @param classLoader the new classloader
-     */
-    public final void setClassLoader(ClassLoader classLoader) {
-        this.classLoader = classLoader;
-    }
-
-    @Override
-    public final void setModuleClassLoader(ClassLoader moduleClassLoader) {
-        this.moduleClassLoader = moduleClassLoader;
-    }
-
-    /**
-     * Sets a named charset.
-     * @param charset the name of a charset
-     * @throws UnsupportedEncodingException if charset is unsupported.
-     */
-    public void setCharset(String charset)
-            throws UnsupportedEncodingException {
-        if (!Charset.isSupported(charset)) {
-            final String message = "unsupported charset: '" + charset + "'";
-            throw new UnsupportedEncodingException(message);
-        }
-        this.charset = charset;
-    }
-
-    /**
-     * Sets the field haltOnException.
-     * @param haltOnException the new value.
-     */
-    public void setHaltOnException(boolean haltOnException) {
-        this.haltOnException = haltOnException;
-    }
+public class Checker extends AbstractChecker {
```

Comment:
Moved `processFilteredFiles` and `processFile` methods to `AbstractChecker` class.

Not sure about `process`, `finishFileSetChecksProcessing` and `destroyFileSetChecks` methods - all of them calls super method